internal/api/api.go

```diff
@@ -0,0 +1,17 @@
+package api
+
+import "github.com/gorilla/mux"
+
+// Register registers the API endpoints on the given router.
+func Register(rootRouter *mux.Router, context *Context) {
+	addContext := func(handler contextHandlerFunc) *contextHandler {
+		return newContextHandler(context, handler)
+	}
+
+	apiRouter := rootRouter.PathPrefix("/api").Subrouter()
+	apiRouter.Handle("/clusters", addContext(handleGetClusters)).Methods("GET")
+	apiRouter.Handle("/clusters", addContext(handleCreateCluster)).Methods("POST")
+	apiRouter.Handle("/cluster/{cluster}", addContext(handleGetCluster)).Methods("GET")
```

Comment:
Three questions here that I'm 0/5 on:

1. Why register the routes here instead of inside of the individual api files (e.g. cluster.go)?
2. Would it make sense to use subrouters here to make it a bit cleaner and easier to extend? Something like:
```go
clustersRouter := apiRouter.PathPrefix("/clusters").Subrouter()
clustersRouter.Handle("", addContext(handleGetClusters)).Methods("GET")
// other "/clusters" routes

clusterRouter := apiRouter.PathPrefix("/cluster/{cluster}").Subrouter()
clusterRouter.Handle("", addContext(handleGetCluster)).Methods("GET")
clusterRouter.Handle("/kubernetes/{version}", addContext(handleUpgradeCluster)).Methods("PUT")
// other "/cluster/{cluster}" routes
```
3. Did you purposefully not use regexes for the path parameters? (e.g. `/cluster/{cluster}:[A-Za-z0-9]+`)

Reply:
1. No objections!
2. Not purposefully :)

Fixed on both counts.